### PR TITLE
Add new case of "Migrate guest after NIC hot-unplug"

### PR DESCRIPTION
**Pull request by kimi1978wy**
Add hotunplug migration for migration_after_nichotplug.py

Signed-off-by: Yu Wang <wyu@redhat.com>

id:1668250

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(1/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (188.18 s)
 (2/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (178.05 s)
 (3/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (179.88 s)
 (4/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (174.29 s)
 (1/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (214.48 s)
 (2/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (194.45 s)
 (3/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000: PASS (193.41 s)
 (4/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (193.84 s)

**Comment by PaulYuuu** [comment]
Test pass. 
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.ppc64le.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (221.82 s)

However, it should be noted that this does not exactly match our manual testing steps.
With hot unplug, steps are:
`boot guest with nic --> hot unplug nic --> migration`

The code execution steps are:
`boot guest without nic --> hot plug nic --> hot unplug nic --> migration`
The final result is not very meaningful.

**Comment by kimi1978wy** [comment]
@PaulYuuu 

> Test pass.
> (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.ppc64le.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (221.82 s)
> 
> However, it should be noted that this does not exactly match our manual testing steps.
> With hot unplug, steps are:
> `boot guest with nic --> hot unplug nic --> migration`
> 
> The code execution steps are:
> `boot guest without nic --> hot plug nic --> hot unplug nic --> migration`
> The final result is not very meaningful.

I asked xuhan, he said it could not support in avocado , so we cannot execute `boot guest with nic --> hot unplug nic --> migration`, I think it covered `unplug --> migration` for this case, and it is the point for this case, so it is meaningful for our test.

**Comment by PaulYuuu** [comment]
> I asked xuhan, he said it could not support in avocado , so we cannot execute `boot guest with nic --> hot unplug nic --> migration`, I think it covered `unplug --> migration` for this case, and it is the point for this case, so it is meaningful for our test.

No, avocado-vt  supports hotunplug nic, but now `def hotunplug_nic()` has some issues. If they are been fixed, We only need to hotunplug it without hotplug.
```
2019-08-16 08:32:42,539 qemu_monitor     L0410 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'device_del'
2019-08-16 08:32:42,540 qemu_monitor     L1920 DEBUG| Send command: {'execute': 'device_del', 'arguments': {'id': 'idPvoVy8'}, 'id': 'AUIuKael'}
2019-08-16 08:32:42,541 qemu_vm          L3672 INFO | waiting for the guest to finish the unplug
...
...
2019-08-16 08:32:47,568 qemu_monitor     L0410 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'netdev_del'
2019-08-16 08:32:47,568 qemu_monitor     L1920 DEBUG| Send command: {'execute': 'netdev_del', 'arguments': {'id': 'idDGafwX'}, 'id': 'Fh0SseI0'}
2019-08-16 08:32:47,570 qemu_monitor     L0410 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'info network' (via Human Monitor)
```
Otherwise, `boot guest without nic --> hot plug nic --> hot unplug nic --> migration` == `boot guest without nic --> migration`

**Comment by kimi1978wy** [comment]
(01/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_8139: PASS (278.65 s)
 (02/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio: PASS (284.72 s)
 (03/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e: PASS (281.89 s)
 (04/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (159.38 s)
 (05/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (159.02 s)
 (06/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (158.92 s)
 (07/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_8139: PASS (222.56 s)
 (08/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio: PASS (227.21 s)
 (09/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000: PASS (227.31 s)
 (10/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e: PASS (223.11 s)
 (11/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (111.31 s)
 (12/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (112.90 s)
 (13/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000: PASS (110.93 s)
 (14/14) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (111.02 s)

**Comment by kimi1978wy** [comment]
@PaulYuuu Updated, pls help to review it again, thanks a lot

**Comment by kimi1978wy** [comment]
(1/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_8139: PASS (230.03 s)
 (2/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio: PASS (229.60 s)
 (3/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000: PASS (231.61 s)
 (4/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e: PASS (230.67 s)
 (5/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (111.65 s)
 (6/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (111.19 s)
 (7/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000: PASS (111.65 s)
 (8/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (111.62 s)

**Comment by kimi1978wy** [comment]
@PaulYuuu Modified, pls help to review it again

File: qemu/tests/cfg/nic_hotplug.cfg [hunk 1]
```diff
@@ -70,8 +70,15 @@
             requires_root = yes
             type = migration_after_nichotplug
             kill_vm = yes
-            nics = ""
```
Comment (PaulYuuu) [inline qemu/tests/cfg/nic_hotplug.cfg]: Hold on, for 'after_nichotplug', we still need this line, otherwise `nics = nic1`
Other LGTM.

**Comment by vivianQizhu** [comment]
Ack.
@kimi1978wy Would you test it with q35? Thanks.

**Comment by kimi1978wy** [comment]
Q35 results:
 (1/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio.q35: PASS (234.99 s)
 (2/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e.q35: PASS (235.60 s)
 (3/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio.q35: PASS (117.05 s)
 (4/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e.q35: PASS (117.37 s)
 (1/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio.q35: PASS (279.71 s)
 (2/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e.q35: PASS (280.34 s)
 (3/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio.q35: PASS (162.79 s)
 (4/4) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e.q35: PASS (160.90 s)

Retest with PC
 (1/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_8139: PASS (223.61 s)
 (2/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_virtio: PASS (223.71 s)
 (3/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000: PASS (226.39 s)
 (4/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotplug.nic_e1000e: PASS (225.97 s)
 (5/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_8139: PASS (112.62 s)
 (6/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_virtio: PASS (112.71 s)
 (7/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000: PASS (112.68 s)
 (8/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.nic_hotplug.migration.after_nichotunplug.nic_e1000e: PASS (112.45 s)

**Pull request merged**